marlowe-cli/src/Language/Marlowe/CLI/Command/Parse.hs

```diff
@@ -359,16 +390,45 @@ parseRole =
 
 
 -- | Parse an address query.
-parseOutputQuery :: O.Parser OutputQuery
+parseOutputQuery :: O.Parser (Maybe (OutputQuery era (OutputQueryResult era)))
 parseOutputQuery =
-  parseAllOutput <|> parseLovelaceOnly <|> parseAssetOnly <|> pure AllOutput
+  Just <$> parseLovelaceOnly <|> Just <$> parseAssetOnly <|> parseAllOutput
     where
       parseAllOutput =
-        AllOutput
+        Nothing
           <$ O.flag' () (O.long "all" <> O.help "Report all output.")
       parseLovelaceOnly =
-        LovelaceOnly . Lovelace
+        LovelaceOnly . (<=) . Lovelace
           <$> O.option O.auto (O.long "lovelace-only" <> O.metavar "LOVELACE" <> O.help "The minimum Lovelace that must be the sole asset in the output value.")
       parseAssetOnly =
         AssetOnly
           <$> O.option parseAssetId (O.long "asset-only" <> O.metavar "CURRENCY_SYMBOL.TOKEN_NAME" <> O.help "The current symbol and token name for the sole native asset in the value.")
+
+
+parseProtocolVersion :: O.ReadM ProtocolVersion
+parseProtocolVersion = O.eitherReader \case
+  "alonzo" -> pure alonzoPV
+  "vasil"  -> pure vasilPV
```

Comment:
It's odd that they call it "vasilPV" instead of "babbagePV".

Reply:
Yeah, but we want to probably to keep it consistent ;-)